src/main/scala/com/monsanto/arch/cloudformation/model/resource/ElasticLoadBalancingV2.scala

```diff
@@ -0,0 +1,408 @@
+  * in the Application Load Balancers Guide.
+  *
+  * @param name CloudFormation logical name
+  * @param DefaultActions The default actions that the listener takes when handling incoming requests.
+  * @param LoadBalancerArn The Amazon Resource Name (ARN) of the load balancer to associate with the listener.
+  * @param Port The port on which the listener listens for requests.
+  * @param Protocol The protocol that clients must use to send requests to the listener.
+  * @param Certificates The SSL server certificate for the listener. With a certificate, you can encrypt traffic between
+  *                     the load balancer and the clients that initiate HTTPS sessions, and traffic between the load
+  *                     balancer and your targets.
+  * @param SslPolicy The security policy that defines the ciphers and protocols that the load balancer supports.
+  * @param Condition Define conditions by using the intrinsic condition functions. These conditions determine when AWS
+  *                  CloudFormation creates the associated resources.
+  */
+case class `AWS::ElasticLoadBalancingV2::Listener`(
+  name:                   String,
+  DefaultActions:         Seq[ListenerAction],
+  LoadBalancerArn:        Token[String],
+  Port:                   Token[Int],
+  Protocol:               ALBProtocol,
+  Certificates:           Option[Seq[Certificate]] = None,
+  SslPolicy:              Option[ELBSecurityPolicy] = None,
+  override val Condition: Option[ConditionRef] = None
+) extends Resource[`AWS::ElasticLoadBalancingV2::Listener`] with HasArn {
+
+  if (Protocol == ALBProtocol.HTTPS && (!Certificates.exists(_.nonEmpty) || SslPolicy.isEmpty))
+    throw new IllegalArgumentException("Certificates and SslPolicy are both required for an HTTPS listener")
+
+  def when(newCondition: Option[ConditionRef] = Condition): `AWS::ElasticLoadBalancingV2::Listener` = copy(Condition = newCondition)
+  def arn: Token[String] = ResourceRef(this)
+}
+
+object `AWS::ElasticLoadBalancingV2::Listener` extends DefaultJsonProtocol {
+  def forHttp(name:            String,
+              DefaultActions:  Seq[ListenerAction],
+              LoadBalancerArn: Token[String],
+              Port:            Token[Int] = 80,
+              Condition:       Option[ConditionRef] = None): `AWS::ElasticLoadBalancingV2::Listener` =
+    `AWS::ElasticLoadBalancingV2::Listener`(
+      name = name,
+      Protocol = ALBProtocol.HTTP,
+      DefaultActions = DefaultActions,
+      LoadBalancerArn = LoadBalancerArn,
+      Port = Port,
+      Condition = Condition
+    )
+
+  def forHttps(name:            String,
+               DefaultActions:  Seq[ListenerAction],
+               LoadBalancerArn: Token[String],
+               Certificates:    Seq[Certificate],
+               Port:            Token[Int] = 443,
+               SslPolicy:       ELBSecurityPolicy = ELBSecurityPolicy.`ELBSecurityPolicy-2016-08`,
```

Comment:
Maybe default this to the current and most secure? `ELBSecurityPolicy-TLS-1-2-2017-01`